src/poetry/inspection/lazy_wheel.py

```diff
@@ -0,0 +1,755 @@
+
+    This wrapper is useful because Python currently does not distinguish read-only
+    streams at the type level.
+    """
+
+    def __init__(self, inner: BinaryIO) -> None:
+        self._file = inner
+
+    @property
+    def mode(self) -> str:
+        """Opening mode, which is always rb."""
+        return "rb"
+
+    @property
+    def name(self) -> str:
+        """Path to the underlying file."""
+        return self._file.name
+
+    def seekable(self) -> bool:
+        """Return whether random access is supported, which is True."""
+        return True
+
+    def close(self) -> None:
+        """Close the file."""
+        self._file.close()
+
+    @property
+    def closed(self) -> bool:
+        """Whether the file is closed."""
+        return self._file.closed
+
+    def fileno(self) -> int:
+        return self._file.fileno()
+
+    def flush(self) -> None:
+        self._file.flush()
+
+    def isatty(self) -> bool:
+        return False
+
+    def readable(self) -> bool:
+        """Return whether the file is readable, which is True."""
+        return True
+
+    def read(self, size: int = -1) -> bytes:
+        """Read up to size bytes from the object and return them.
+
+        As a convenience, if size is unspecified or -1,
+        all bytes until EOF are returned.  Fewer than
+        size bytes may be returned if EOF is reached.
+        """
+        return self._file.read(size)
+
+    def readline(self, limit: int = -1) -> bytes:
+        # Explicit impl needed to satisfy mypy.
+        raise NotImplementedError
+
+    def readlines(self, hint: int = -1) -> list[bytes]:
+        raise NotImplementedError
+
+    def seek(self, offset: int, whence: int = 0) -> int:
+        """Change stream position and return the new absolute position.
+
+        Seek to offset relative position indicated by whence:
+        * 0: Start of stream (the default).  pos should be >= 0;
+        * 1: Current position - pos may be negative;
+        * 2: End of stream - pos usually negative.
+        """
+        return self._file.seek(offset, whence)
+
+    def tell(self) -> int:
+        """Return the current position."""
+        return self._file.tell()
+
+    def truncate(self, size: int | None = None) -> int:
+        """Resize the stream to the given size in bytes.
+
+        If size is unspecified resize to the current position.
+        The current stream position isn't changed.
+
+        Return the new file size.
+        """
+        return self._file.truncate(size)
+
+    def writable(self) -> bool:
+        """Return False."""
+        return False
+
+    def write(self, s: Any) -> int:
+        raise NotImplementedError
+
+    def writelines(self, lines: Iterable[Any]) -> None:
+        raise NotImplementedError
+
+    def __enter__(self) -> ReadOnlyIOWrapper:
+        self._file.__enter__()
+        return self
+
+    def __exit__(self, *exc: Any) -> None:
+        self._file.__exit__(*exc)
```

Comment:
```suggestion
    def __exit__(
        self,
        exc_type: type[BaseException] | None,
        exc_value: BaseException | None,
        traceback: TracebackType | None,
    ) -> bool:
    	return self._file.__exit__(exc_type, exc_value, traceback)
```
(`TracebackType` requires importing from `types`)

same applies to L301

Reply:
That does not work for me. mypy fails if I try this.

Reply:
This should work then:
```suggestion
    def __exit__(
        self,
        exc_type: type[BaseException] | None,
        exc_value: BaseException | None,
        traceback: TracebackType | None,
    ) -> None:
    	self._file.__exit__(exc_type, exc_value, traceback)
```
Seems like Python itself doesn't exactly follow the standard signature 😄

Reply:
That works. 👍